index.js

```diff
@@ -1 +1,104 @@
 // implement your API here
+
+const express = require('express');
+
+const db = require('./data/db');
+
+const server = express();
+
+server.use(express.json());
+
+// GET
+
+server.get('/api/users', (req, res) => {
+    db.find()
+        .then(users => {
+            res.json(users)
+        })
+        .catch(err => {
+            res.status(500).json({ error: err, message: 'The users information could not be retrieved'})
+        })
+})
+
+// GET by id
+
+server.get('/api/users/:id', (req, res) => {
+    const userId = req.params.id;
+
+    db.findById(userId)
+        .then(users => {
+            res.json(users)
+        })
+        .catch(err => {
+            res.status(404).json({ error: err, message: 'The user with the specified ID does not exist.' })
+        })
+})
+
+// POST
+
+server.post('/api/users', (req, res) => {
+    const newUser = req.body;
+
+    if (!req.body.name && !req.body.bio) {
```

Comment:
this should be an OR statement instead of an AND, because if either of the two does not present then you want to throw an error.